Update to v9.0.1 released 2025-02-11

bmkramer opened the pull request:
- updated Readme (table counts + Zenodo links) 
- updated schema for each table type to v9.0.1

bmkramer committed:
Update Readme and config.yaml for v9.0.1 (Feb 2025)

bmkramer committed:
update schemas to v9.0.1.

bmkramer committed:
edit schema

bmkramer committed:
add product.json as scheme for all research product files (publication, dataset, software, otherresearchproduct) when importing from GCS into GBQ via web UI.

kathrynnapier commented:
Hi @keegansmith21 - not urgent, but when you have some time, could you please review this?

keegansmith21 commented:
Thanks @bmkramer 
Is the product table new? If so it looks like it will need to be added to the config file.

bmkramer committed:
rename results schema to product schema to match files names in Zenodo record

bmkramer commented:
Hi @keegansmith21 It's new, but might only be relevant for manual upload (which I used this time). In the workflow, the respective schema's for publication, dataset, software and otherresearchproduct are used, not the umbrella schema which is the product table. 

I added it because with manual upload, it's convenient to use the full schema. Variables not included in a particular table appear to be safely nulled, while variables in the dataset but not in the schema give an error, which is helpful to make sure all variables are included in the schema. 

In the workflow, this might not work and give errors when variables are in the scheme and not in the data (I did not test this for this version though!). So for the workflow, it might be best to keep using the individual schema's (which I also updated).

keegansmith21 commented:
No worries. In that case it all looks good to me. Thanks @bmkramer

bmkramer committed:
Update datasource.json

add contentTypes
